File: src/Java/StructuralPatterns/Composite/WindowsFormsApplication/src/Component/Control.java
package Component;

import java.awt.Color;
import java.util.List;

public abstract class Control {
    private final String name;
    private final int width;
    private final int height;
    private final Color foreColor;
    private final Color backColor;

    public Control(String name, int width, int height, Color foreColor, Color backColor) {
        this.name = name;
        this.width = width;
        this.height = height;
        this.foreColor = foreColor;
        this.backColor = backColor;
    }

    public String getName() {
        return name;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public Color getForeColor() {
        return foreColor;
    }

    public Color getBackColor() {
        return backColor;
    }

    public abstract List<Control> getControls();

    public abstract void add(Control control);

    public abstract void remove(Control control);

    public abstract String getHierarchicalLevel(int depthIndicator);
}

File: src/C#/StructuralPatterns/Composite/WindowsFormsApplication/Component/Control.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplication.Component
{
    public abstract class Control
    {
        public Control(string name, int width, int height, Color foreColor, Color backColor)
        {
            Name = name;
            Width = width;
            Height = height;
            ForeColor = foreColor;
            BackColor = backColor;
        }

        public string Name { get; set; }

        public int Width { get; set; }

        public int Height { get; set; }

        public Color ForeColor { get; set; }

        public Color BackColor { get; set; }

        public abstract ICollection<Control> Controls { get; set; }

        public abstract void Add(Control control);

        public abstract void Remove(Control control);

        public abstract string GetHierarchicalLevel(int depthIndicator);
    }
}
